bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -824,323 +855,181 @@ def get_in_progress_submission(user: User, uuid: str) -> Optional[InProgressSubm
-            if session:
-                session_check = _accept_session(filename, filesize, session)
-                if not session_check["accepted"]:
-                    return JsonResponse(session_check, status=200)
-
-        # The contents of the file are not known here, so it is not necessary to
-        # call _accept_content()
+        uploaded_file = UploadedFile(session=session, file_upload=_file, name=_file.name)
+        uploaded_file.save()
+        file_url = uploaded_file.get_file_access_url()
 
-        return JsonResponse({"accepted": True}, status=200)
+        return JsonResponse(
+            {
+                "file": _file.name,
+                "accepted": True,
+                "uploadSessionToken": session.token,
+                "url": file_url,
+            },
+            status=200,
+        )
 
     except Exception as exc:
-        LOGGER.error(msg=("Uncaught exception in checkfile view: {0}".format(str(exc))))
+        LOGGER.error(msg=("Uncaught exception in upload_file view: %s", exc))
         return JsonResponse(
             {
-                "accepted": False,
-                "error": gettext("500 Internal Server Error"),
-                "verboseError": gettext("500 Internal Server Error"),
+                "error": gettext("There was an internal server error. Please try again."),
             },
             status=500,
         )
 
 
-def _accept_file(filename: str, filesize: Union[str, int]) -> dict:
-    """Determine if a new file should be accepted. Does not check the file's
-    contents, only its name and its size.
-
-    These checks are applied:
-    - The file name is not empty
-    - The file has an extension
-    - The file's extension exists in ACCEPTED_FILE_FORMATS
-    - The file's size is an integer greater than zero
-    - The file's size is less than or equal to the maximum allowed size for one file
+@require_http_methods(["GET"])
+def list_uploaded_files(request: HttpRequest, session_token: str) -> JsonResponse:
+    """Get a list of metadata for files that have been uploaded in a given upload session.
 
     Args:
-        filename (str): The name of the file
-        filesize (Union[str, int]): A string or integer representing the size of
-            the file (in bytes)
+        request: The HTTP request
+        session_token: The upload session token from the URL
 
     Returns:
-        (dict): A dictionary containing an 'accepted' key that contains True if
-            the session is valid, or False if not. The dictionary also contains
-            an 'error' and 'verboseError' key if 'accepted' is False.
+        JsonResponse: A JSON response containing the list of uploaded files and their details,
+        or an error message if the session is not found.
     """
+    session = UploadSession.objects.filter(token=session_token).first()
+    if not session:
+        # Create a new session if one is not found
+        session = UploadSession.new_session()
+        session.save()
```

Comment:
Is there a need for this behaviour? A GET request shouldn't create resources on the backend. I don't think the uploadSessionToken from this request is even being used.